src/Parser/Shaman/Elemental/Modules/Items/TheDeceiversBloodPact.js

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import ITEMS from 'common/ITEMS';
+import SPELLS from 'common/SPELLS';
+import ItemIcon from 'common/ItemIcon';
+import ItemLink from 'common/ItemLink';
+import { formatNumber } from 'common/format';
+
+
+class TheDeceiversBloodPact extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+  };
+
+  extraMaelstrom = 0;
+
+  on_initialized() {
+    this.active = this.combatants.selected.hasBuff(SPELLS.THE_DECEIVERS_BLOOD_PACT_EQUIP.id);
+  }
+
+  on_byPlayer_energize(event) {
+    if (event.ability.guid===SPELLS.THE_DECEIVERS_BLOOD_PACT_BUFF.id) {
+        this.extraMaelstrom+=event.classResources[0].amount;
```

Comment:
For Shaman I believe this is fine to code a index `0` just for other classes, such as druids, their main resource potentially changes.